### PR TITLE
Imviz: Make the WCS match tool into a pan/zoom with WCS matching tool

### LINKED ISSUE #519: JDAT-1221: WCS and Pixel Matching and Locking
Lock selected viewers to show the same area in WCS or pixel coordinates.

UI/UX:

1. Match and/or Lock all viewers or selected viewers
2. Separate icon (kabob) for blinking/locking images

https://jira.stsci.edu/browse/JDAT-1221

# Dev notes

@astrofrog and @pllim agreed on the following behaviors:

* When zooming, matching viewers will not update until zoom is completed (when scroll stops).
* When panning, matching viewers will not update until pan is done (mouse button is released).

### PR DESCRIPTION
**Pull request by astrofrog**
This PR changes the existing WCS matching tool to be a Pan/Zoom tool which matches WCS at the end of the pan/zoom as described in https://github.com/spacetelescope/jdaviz/issues/519.

A few high-level decisions are needed:

* For now, the datasets are linked on the fly when the tool is enabled, but this seems ugly, so I guess the question is whether we want to always link WCSes by default, either in jdaviz in general or in imviz. Is there any reason we would not want images linked by WCS in imviz? The linking doesn't mean the axes have to match on all image viewers, but more that glue will automatically know how to e.g. overlay datasets and so on and will allow blinking (since that also requires the WCSes to be matched). I personally think that imviz should always link datasets by WCS if present, but happy to hear counter-argumnents (we don't have to do this for all of jdaviz though)

* Clicking on the WCS pan/zoom tool will automatically cause images in other viewers to be registered to the image in the viewer where the tool is enabled. This seems sensible, but what should users have to do if they want to get back to a view that is not matched? Should the 'home' icon do this?

Here's a preview of the current functionality (sorry for the bit where the viewer scrolls out of view, this is the bug fixed by @maartenbreddels in bqplot but I wasn't using that version):

ezgif com-gif-maker

Closes #519

### CODE REVIEW

File: jdaviz/configs/imviz/plugins/viewers.py
Comment (astrofrog) [inline jdaviz/configs/imviz/plugins/viewers.py:19]: @pllim - just FYI this is why the home button was inherited by default before. I think we don't actually want the inheritance here because who knows what tools will get added in glue-jupyter in future so I have disabled it for imviz. We should perhaps do this for other jdaviz apps but this is beyond the scope of this PR.

**Comment by astrofrog** [comment]
(this should be rebased once #614 is merged)

**Comment by pllim** [comment]
> always link WCSes by default, either in jdaviz in general or in imviz

Just doing it in Imviz is probably less controversial. I have no idea how other viz is set up right now. Does this mean, we can remove the "link" button then?

> automatically cause images in other viewers to be registered to the image in the viewer where the tool is enabled

I guess this satisfies the "or" clause under, "Match and/or Lock all viewers or selected viewers." Not sure how you select viewers but maybe that can be future work.

> Should the 'home' icon do this?

Yes, that makes sense to me.

File: jdaviz/data/icons/convert.sh
Comment (pllim) [inline jdaviz/data/icons/convert.sh:3]: What is this for? Is this executed at run time?

Comment (astrofrog) [reply jdaviz/data/icons/convert.sh:3]: It's just so that if we add more svg icons we can convert them all in a consistent way. It's not meant to be part of the installed package. Maybe I can move it to the root and call it ``.convert_icons_to_png.sh`` or something

Comment (pllim) [reply jdaviz/data/icons/convert.sh:3]: It is only one line of command. Can we just move it to https://github.com/spacetelescope/jdaviz/wiki ?

Comment (astrofrog) [reply jdaviz/data/icons/convert.sh:3]: True, my point was more that it might grow as more icons are added and we don't want all icons to be added in an inconsistent way. What about putting it as a comment in the ``__init__.py`` of the icons directory?

Comment (pllim) [reply jdaviz/data/icons/convert.sh:3]: Not sure how discoverable that is. How about a short paragraph in dev docs then?

**Comment by pllim** [comment]
#614 is merged. Please rebase, @astrofrog . Thanks!

**Comment by PatrickOgle** [comment]
Yes there are reasons not to link the WCS in separate viewers.  I might want to keep a larger FOV in one window and a zoom only into one of the viewers.  The lock function should be toggle-able.

**Comment by PatrickOgle** [comment]
What happens when you push the button a 3rd time and a 3rd viewer appears, both in terms of the locking capabililty and how the viewers are arranged on the screen.  In a grid?

**Comment by pllim** [comment]
Let's not let perfect be the enemy of good. Can we have the basic functionality wrapped up here and discuss auto-linking in a separate issue/PR? Whether initial default is "link all" or "opt in only," I'll leave that to y'all to fight it out.

**Comment by astrofrog** [comment]
> Yes there are reasons not to link the WCS in separate viewers. I might want to keep a larger FOV in one window and a zoom only into one of the viewers. The lock function should be toggle-able.

To clarify, this is how this behaves at the moment. Auto-linking data in glue won't cause viewers to always be locked - linking is just a way for glue to understand the connection between datasets, but I fully intend to have the pan/zoom locking be opt-in.

File: jdaviz/data/icons/convert.sh
Comment (astrofrog) [reply jdaviz/data/icons/convert.sh:3]: Actually let me just remove it - we really should just get the SVG icons to work.

Comment (astrofrog) [reply jdaviz/data/icons/convert.sh:3]: Which I'll look into in a separate PR but my point is we don't need to document how to get the SVG.

**Comment by astrofrog** [comment]
This is now rebased and to be clear the linking only happens when the tool is toggled. I'll open a separate PR about auto-linking.

**Comment by pllim** [comment]
The `alldeps` failure is annoying and appears unrelated. I have asked `jwst` devs. Unfortunately, it blocks the rest of the CI, though I see little reason why CI would fail with this patch. 

tl;dr -- Ignore the CI failure.

**Review by pllim — APPROVED** [review]
Works as advertised with one caveat: If I zoom first on the reference image and then click on the "arrow with linking symbol", the other image zoom does not automatically update to reference image zoom.

**Comment by pllim** [comment]
The slight lag is also unfortunate but is out of scope here (see #564).

**Comment by astrofrog** [comment]
@pllim - the latest commit fixes that small issue you noticed - now when clicking on the tool the initial limits will be matched.

**Review by pllim — APPROVED** [review]
Works great now, both on "standalone" and notebook form. Didn't even break when I activate the "arrows with linking" on multiple sub-viewers. There is still some lag, but like I said, that is a separate known issue.

### Standalone

Screenshot 2021-05-25 104930

### Classic notebook

Screenshot 2021-05-25 105936

**Comment by pllim** [comment]
@astrofrog , if you rebase, the CI should pass now, unless you broke it here yourself. 😹

**Comment by astrofrog** [comment]
@pllim - rebased!

**Review by duytnguyendtn — APPROVED** [review]
Tested in both the desktop and notebook interface. They look good to me!

**Comment by pllim** [comment]
Still works after rebase. I will merge when CI passes.

**Comment by pllim** [comment]
Thanks!

**Pull request merged**
